OneSignalSDK/onesignal/notifications/src/main/java/com/onesignal/notifications/activities/NotificationOpenedActivityBase.kt

```diff
@@ -46,25 +46,17 @@ abstract class NotificationOpenedActivityBase : Activity() {
     }
 
     internal open fun processIntent() {
-        if (!OneSignal.initWithContext(applicationContext)) {
-            return
+        suspendifyOnThread {
+            if (!OneSignal.initWithContext(applicationContext)) {
+                return@suspendifyOnThread
+            }
+
+            val openedProcessor = OneSignal.getService<INotificationOpenedProcessor>()
+            openedProcessor.processFromContext(this, intent)
+
+            runOnUiThread {
+                AndroidUtils.finishSafely(this)
+            }
         }
-        suspendifyOnThread(
-            block = {
-                val openedProcessor = OneSignal.getService<INotificationOpenedProcessor>()
-                openedProcessor.processFromContext(this, intent)
-                // KEEP: Xiaomi Compatibility:
-                // Must keep this Activity alive while trampolining, that is
-                // startActivity() must be called BEFORE finish(), otherwise
-                // the app is never foregrounded.
-            },
-            onComplete = {
-                // Safely finish the activity on the main thread after processing is complete.
-                // This gives the system enough time to complete rendering before closing the Trampoline activity.
```

Comment:
Please keep these comments, it's important so authors of future changes don't break Xiaomi compatibility, it is very easy to do so and very hard to debug if it happens.